File: backend/core/roles.py
"""The seven platform roles."""

from django.db import models


class Role(models.TextChoices):
    SUPER_ADMIN = "super_admin", "Super Admin"
    ADMIN = "admin", "Admin"
    MIS = "mis", "MIS Executive"
    COUNSELOR = "counselor", "Counselor"
    TECH_SUPPORT = "tech_support", "Tech Support"
    FACULTY = "faculty", "Faculty"
    STUDENT = "student", "Student"


# Roles that sign in through a staff login page (everyone except Student).
STAFF_ROLES = frozenset(
    {
        Role.SUPER_ADMIN,
        Role.ADMIN,
        Role.MIS,
        Role.COUNSELOR,
        Role.TECH_SUPPORT,
        Role.FACULTY,
    }
)

File: frontend/src/app/roles.ts
export interface RoleDef {
  /** URL segment for the login + portal route. */
  slug: string;
  /** Backend role value (matches core.roles.Role). */
  value: string;
  label: string;
  tagline: string;
}

/** The seven roles, each with its own login page (role-bound). */
export const ROLES: RoleDef[] = [
  { slug: "student", value: "student", label: "Student", tagline: "Your batch, in one place." },
  {
    slug: "faculty",
    value: "faculty",
    label: "Faculty",
    tagline: "Teach, grade, and guide your batches.",
  },
  { slug: "admin", value: "admin", label: "Admin", tagline: "Run day-to-day operations." },
  { slug: "mis", value: "mis", label: "MIS Executive", tagline: "Monitor sessions and escalations." },
  { slug: "counselor", value: "counselor", label: "Counselor", tagline: "Attendance and follow-up." },
  {
    slug: "tech-support",
    value: "tech_support",
    label: "Tech Support",
    tagline: "Keep the doubt forum moving.",
  },
  {
    slug: "super-admin",
    value: "super_admin",
    label: "Super Admin",
    tagline: "Full control and settings.",
  },
];

/** Map a backend role value to its route slug (e.g. tech_support -> tech-support). */
export function slugForRole(value: string): string {
  return ROLES.find((r) => r.value === value)?.slug ?? "";
}
